### ISSUE TITLE
Property's model name prefix and duplication

### ISSUE THREAD

**Issue by Tomas2D** [koxudaxi/datamodel-code-generator#1737]
**Is your feature request related to a problem? Please describe.**
Currently, when the parser finds two properties with the same name, it appends the number at the end to prevent duplication. 

```python
class Result(BaseModel):
   ...

class UserListResponse(BaseModel):
  result: list[Result]

class Result2(BaseModel):
   ...

class UserDetailResponse(BaseModel):
  result: Result2
```

**Describe the solution you'd like**

I would like to have the option to use the parent's model name as a prefix for the property name.

```python
class UserListResponseResult(BaseModel):
   ...

class UserListResponse(BaseModel):
  result: list[UserListResponseResult]

class UserDetailResponseResult(BaseModel):
   ...

class UserDetailResponse(BaseModel):
  result: UserDetailResponseResult
```

**Comment by Tomas2D** [comment]
Related: #796

**Comment by SigveMartin** [comment]
@Tomas2D did you figure out a solution for this? I have a similar case, and looked for the same solution as you describe above.

**Comment by kylebebak** [comment]
Agreed that a fix for this issue would be great. I think this issue is basically a dupe of #796

**Comment by Tomas2D** [comment]
@SigveMartin yes 

Source: https://github.com/IBM/ibm-generative-ai/tree/main/scripts/types_generator
Aliases File: https://github.com/IBM/ibm-generative-ai/blob/main/scripts/types_generator/schema_aliases.yaml

**Comment by gaborbernat** [comment]
> Agreed that a fix for this issue would be great. I think this issue is basically a dupe of [#796](https://github.com/koxudaxi/datamodel-code-generator/issues/796)
